clippy_lints/src/methods/io_other_error.rs

```diff
@@ -0,0 +1,31 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::msrvs::{IO_ERROR_OTHER, Msrv};
+use rustc_errors::Applicability;
+use rustc_hir::Expr;
+use rustc_lint::LateContext;
+
+pub(super) fn check(cx: &LateContext<'_>, expr: &Expr<'_>, path: &Expr<'_>, args: &[Expr<'_>], msrv: &Msrv) {
+    if let [error_kind, error] = args
+        && !expr.span.from_expansion()
+        && !error_kind.span.from_expansion()
+        && clippy_utils::is_expr_path_def_path(cx, path, &clippy_utils::paths::IO_ERROR_NEW)
+        && clippy_utils::is_expr_path_def_path(
+            cx,
+            clippy_utils::expr_or_init(cx, error_kind),
+            &clippy_utils::paths::IO_ERRORKIND_OTHER,
+        )
+        && msrv.meets(IO_ERROR_OTHER)
+    {
+        let mut applicability = Applicability::MachineApplicable;
+        let snip = clippy_utils::source::snippet_with_applicability(cx, error.span, "_", &mut applicability);
+        span_lint_and_sugg(
+            cx,
+            super::IO_OTHER_ERROR,
+            expr.span,
+            "this can be `io::Error::other`",
+            "use",
+            format!("std::io::Error::other({snip})"),
```

Comment:
I find it a bit suboptimal that it suggests the full `std::io::Error` path even if a user might have imported it. Couldn't we snipp this part from the `path` expression?

Reply:
I agree that it's suboptimal, but I just tried `with_forced_trimmed_path!` and it didn't work. I'll have a look around to see whether there's a solution.

Reply:
Thanks for investigating, if you don't find a simple solution, let's not block the PR on this :)